index.js

```diff
@@ -106,12 +115,27 @@ class Logger {
             this.command += CONFIG.SYSTEM.reset;
         }
         console.log(this.command);
+
+        // Writing the previous command to a file if the current command is not a continuation of the previous command
+        if (!stripAnsi(this.command).startsWith(stripAnsi(this.lastCommand))) {
+            this.saveLogToFile()
+        }
         // Save last command if we need to use for joint
         this.lastCommand = this.command;
         this.command = '';
         return this;
     }
 
+    setLogFile(logFilePath) {
+        // Create a stream to save log messages
+        this.logFileStream = createWriteStream(resolve(logFilePath), { flags: 'a' });
+    }
+
+    saveLogToFile() {
```

Comment:
no need this function. pipe the command to the stream every single time while we handling the incoming command.

Reply:
If incoming messages are written immediately to the stream, duplicates will appear when using joint(), so a delayed sending approach is used.

Reply:
sure, It should be correct to redirect the output wherever  `console.log()` is called in the current source code.

Reply:
Okay, I'll try to implement this in the next couple days.

Reply:
sure, let me know if you have any questions